Branch-SDK/src/io/branch/referral/ExtendedAnswerProvider.java

```diff
@@ -0,0 +1,152 @@
+package io.branch.referral;
+
+import android.text.TextUtils;
+import android.util.Log;
+
+import com.crashlytics.sdk.android.answersshim.AnswersOptionalLogger;
+import com.crashlytics.sdk.android.answersshim.KitEvent;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.Iterator;
+
+/**
+ * Created by sojanpr on 6/8/16.
+ * <p>
+ * Class for providing extended data to Fabric Answers.
+ * Create Kit events for the "install" / "open" / "share" events from Branch and update the Answers through
+ * Answers-Shim API. Class handles the logic for flattening  a Branch request to discrete key-value pairs for Answer {@link KitEvent}.
+ * </p>
+ */
+class ExtendedAnswerProvider {
+    // Event Name for Kit Event
+    private static final String KIT_EVENT_INSTALL = "install";
+    private static final String KIT_EVENT_OPEN = "open";
+    private static final String KIT_EVENT_SHARE = "share";
+
+    private static final String EXTRA_PARAM_NOTATION = "+";
+    private static final String CTRL_PARAM_NOTATION = "~";
+    private static final String INNER_PARAM_NOTATION = ".";
+
+    /**
+     * <p>
+     * Method for sending the kit events to Answers for Branch Events.
+     * Method create a Json Object to flatten and create the {@link KitEvent} with key-values
+     * </p>
+     *
+     * @param request  {@link ServerRequest} corresponding to the event
+     * @param response {@link ServerResponse} corresponding to the event
+     */
+    public static void provideData(ServerRequest request, ServerResponse response) {
```

Comment:
Can you just pass in the event name like Ahmed does [here[(https://github.com/BranchMetrics/ios-branch-deep-linking/pull/367/files)? You have that context in every original call? By passing in a server request, you're creating a dependency of this class on external code structure.